src/main/java/wraith/fwaystones/mixin/FilledMapItemMixin.java

```diff
@@ -0,0 +1,22 @@
+package wraith.fwaystones.mixin;
+
+import com.llamalad7.mixinextras.injector.ModifyReceiver;
+import net.minecraft.item.FilledMapItem;
+import net.minecraft.item.map.MapState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.WorldAccess;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import wraith.fwaystones.item.map.MapStateAccessor;
+
+@Mixin(FilledMapItem.class)
+public class FilledMapItemMixin {
+    @ModifyReceiver(method = "updateColors", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/map/MapState;removeBanner(Lnet/minecraft/world/BlockView;II)V"))
+    private MapState removeWaystones(MapState receiver, BlockView world, int x, int y) {
+        BlockPos removalPos;
+        if ((removalPos = ((MapStateAccessor)receiver).removeWaystone(world, x, y)) != null)
+            ((MapStateAccessor)receiver).addWaystone((WorldAccess) world, removalPos);
```

Comment:
Is this supposed to be ***add***Waystone?

Reply:
Again, this is the way minecraft does it for banners, I just copied minecraft's code here because I thought it would be better to share the same behaviour for both banners and waystones.

Reply:
> Is this supposed to be _**add**_Waystone?

Yes and no, I should've probably named it as "updateWaystones", since it deletes the current waystone marker, and attempts to add it again. Without these lines, waystone names won't change on map if changed in the world and they won't be removed if destroyed.